botstrap.py

```diff
@@ -65,64 +90,113 @@ class DiscordClient(Client):
+        app = self.app_info
+        current_flags = app.get("flags", 0)
+        new_flags = current_flags | 1 << 15 | 1 << 19
+        if new_flags != current_flags:
+            log.info("Upgrading application flags to enable required intents")
+            resp = self.patch("/applications/@me", json={"flags": new_flags})
+            self._app_info = cast("dict[str, Any]", resp.json())
+            return True
+        return False
+
+    def check_if_in_guild(self) -> bool:
+        """Check if the bot is a member of the guild."""
+        try:
+            _ = self.guild_info
+        except Exception:
+            return False
+        return True
+
     def upgrade_server_to_community_if_necessary(
         self,
         rules_channel_id_: int | str,
         announcements_channel_id_: int | str,
     ) -> None:
         """Fetches server info & upgrades to COMMUNITY if necessary."""
-        response = self.get(f"/guilds/{self.guild_id}")
-        payload = response.json()
+        payload = self.guild_info
 
         if COMMUNITY_FEATURE not in payload["features"]:
-            log.warning("This server is currently not a community, upgrading.")
+            log.info("This server is currently not a community, upgrading.")
             payload["features"].append(COMMUNITY_FEATURE)
             payload["rules_channel_id"] = rules_channel_id_
             payload["public_updates_channel_id"] = announcements_channel_id_
-            self.patch(f"/guilds/{self.guild_id}", json=payload)
+            self._guild_info = self.patch(f"/guilds/{self.guild_id}", json=payload).json()
             log.info(f"Server {self.guild_id} has been successfully updated to a community.")
 
-    def create_forum_channel(
-        self,
-        channel_name_: str,
-        category_id_: int | str | None = None
-    ) -> int:
+    def create_forum_channel(self, channel_name_: str, category_id_: int | str | None = None) -> str:
         """Creates a new forum channel."""
         payload = {"name": channel_name_, "type": GUILD_FORUM_TYPE}
         if category_id_:
             payload["parent_id"] = category_id_
 
-        response = self.post(f"/guilds/{self.guild_id}/channels", json=payload)
+        response = self.post(
+            f"/guilds/{self.guild_id}/channels",
+            json=payload,
+            headers={"X-Audit-Log-Reason": f"PyDis Botstrap: Creating {channel_name_} forum channel"},
+        )
         forum_channel_id = response.json()["id"]
         log.info(f"New forum channel: {channel_name_} has been successfully created.")
         return forum_channel_id
 
     def is_forum_channel(self, channel_id_: str) -> bool:
         """A boolean that indicates if a channel is of type GUILD_FORUM."""
-        response = self.get(f"/channels/{channel_id_}")
-        return response.json()["type"] == GUILD_FORUM_TYPE
+        return next(filter(lambda c: c["id"] == channel_id_, self.guild_channels)).get("type", None) == GUILD_FORUM_TYPE
```

Comment:
This isn't very clean and channel `type` is a mandatory attribute from the API, split it up and remove the unnecessary `get()` usage.

I suggest a new helper method to return a channel from `self.guild_channels` and then this method just has to use that and check the type key.

I'd also remove the unnecessary `_` suffixes from the arguments to these helper methods.

Reply:
To be fair, given the template now has the forum channel in it, this could be removed entirely. In my testing the code that upgrades python-help doesn't run when a guild is created from the template.